gc/default/default.c

```diff
@@ -7541,34 +7541,34 @@ rb_gc_impl_stat(void *objspace_ptr, VALUE hash_or_sym)
     SET(sweeping_time, (size_t)ns_to_ms(objspace->profile.sweeping_time_ns));
 
     /* implementation dependent counters */
+    SET(major_gc_count, objspace->profile.major_gc_count);
+    SET(minor_gc_count, objspace->profile.minor_gc_count);
+    SET(total_allocated_pages, objspace->heap_pages.allocated_pages);
+    SET(total_freed_pages, objspace->heap_pages.freed_pages);
     SET(heap_allocated_pages, rb_darray_size(objspace->heap_pages.sorted));
+    SET(heap_eden_pages, heap_eden_total_pages(objspace));
     SET(heap_empty_pages, objspace->empty_pages_count)
-    SET(heap_allocatable_slots, objspace->heap_pages.allocatable_slots);
     SET(heap_available_slots, objspace_available_slots(objspace));
     SET(heap_live_slots, objspace_live_slots(objspace));
     SET(heap_free_slots, objspace_free_slots(objspace));
     SET(heap_final_slots, total_final_slots_count(objspace));
     SET(heap_marked_slots, objspace->marked_slots);
-    SET(heap_eden_pages, heap_eden_total_pages(objspace));
-    SET(total_allocated_pages, objspace->heap_pages.allocated_pages);
-    SET(total_freed_pages, objspace->heap_pages.freed_pages);
+    SET(heap_allocatable_slots, objspace->heap_pages.allocatable_slots);
     SET(total_allocated_objects, total_allocated_objects(objspace));
```

Comment:
I think I would put the `total_*object` keys before the pages one because they are part of the more common usage.

Pages stats are for much more advanced usage, as you need to know pages sizes etc to make sense of them.